CompoundV3-Bots/Reserves-Monitor/src/agent.spec.ts

```diff
@@ -0,0 +1,337 @@
+import {
+  FindingType,
+  FindingSeverity,
+  Finding,
+  HandleBlock,
+  BlockEvent,
+} from "forta-agent";
+import { RESERVES_ABI, TARGET_RESERVES_ABI } from "./constants";
+
+import {
+  createAddress,
+  TestBlockEvent,
+  MockEthersProvider,
+} from "forta-agent-tools/lib/tests";
+import { BigNumber } from "ethers";
+import { Interface } from "ethers/lib/utils";
+import Fetcher from "./dataFetcher";
+
+const TEST_ADDRESSES = [createAddress("0x11"), createAddress("0x12")];
+const mockProvider: MockEthersProvider = new MockEthersProvider();
+const ALERT_FREQ = 1000;
+const mockGetFn = (id: string) => {
+  if (id == "cometAddresses") return TEST_ADDRESSES;
+  else return ALERT_FREQ;
+};
+const mockNetworkManager = {
+  cometAddresses: TEST_ADDRESSES,
+  networkMap: {},
+  setNetwork: jest.fn(),
+  get: mockGetFn as any,
+};
```

Comment:
You can also simply use the normal `NetworkManager` with a test config and initialize it using the mock provider. Something like (inside `describe`):

```TS
  let provider: ethers.providers.Provider;
  let mockProvider: MockEthersProvider;
  let networkManager: NetworkManager<NetworkData>;
  let handleBlock: HandleBlock;

  beforeEach(async () => {
    mockProvider = new MockEthersProvider();
    mockProvider.setNetwork(network);
    provider = mockProvider as unknown as ethers.providers.Provider;

    networkManager = new NetworkManager(DEFAULT_CONFIG);
    
    const initialize = provideInitialize([...]);
    await initialize();

    handleBlock = provideHandleBlock([...]);
  });
  ```